sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java

```diff
@@ -67,20 +67,58 @@ public ScannerPluginInstaller(ScannerWsClient wsClient, FileCache fileCache, Sca
   private Map<String, ScannerPlugin> loadPlugins(InstalledPlugin[] remotePlugins) {
     Map<String, ScannerPlugin> infosByKey = new HashMap<>(remotePlugins.length);
 
-    Profiler profiler = Profiler.create(LOG).startDebug("Load plugins");
+    Profiler profiler = Profiler.create(LOG).startInfo("Load/download plugins");
+    Progress progress = new Progress();
 
-    for (InstalledPlugin installedPlugin : remotePlugins) {
-      if (pluginPredicate.apply(installedPlugin.key)) {
-        File jarFile = download(installedPlugin);
-        PluginInfo info = PluginInfo.create(jarFile);
-        infosByKey.put(info.getKey(), new ScannerPlugin(installedPlugin.key, installedPlugin.updatedAt, info));
+    if (!LOG.isDebugEnabled()) {
+      progress.start();
+    }
+    try {
+      for (InstalledPlugin installedPlugin : remotePlugins) {
+        if (pluginPredicate.apply(installedPlugin.key)) {
+          File jarFile = download(installedPlugin);
+          PluginInfo info = PluginInfo.create(jarFile);
+          infosByKey.put(info.getKey(), new ScannerPlugin(installedPlugin.key, installedPlugin.updatedAt, info));
+        }
       }
+    } finally {
+      progress.flagStop();
     }
 
-    profiler.stopDebug();
+    profiler.stopInfo();
     return infosByKey;
   }
 
+  private static class Progress extends Thread {
+    private volatile boolean stop = false;
+
+    private Progress() {
+      super("plugin-load-progress");
+    }
+
+    @Override
+    public void run() {
+      while (!stop) {
+        try {
+          System.out.print('.');
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
+      System.out.println();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)